src/sentry/static/sentry/app/utils/discover/fields.tsx

```diff
@@ -757,6 +784,12 @@ function validateForNumericAggregate(
   };
 }
 
+function validateAllowedColumns(validColumns: string[]): ValidateColumnFunction {
+  return function ({name}: {name: string}): boolean {
+    return validColumns.includes(name);
+  };
+}
```

Comment:
You should be able to just change `ValidateColumnFunction` to `ValidateColumnValueFunction` and remove the extra types in the result here. This is because the return type of `validateAllowedColumns` already enforces the type of the returned function. Then we can simply not use the `datatType` field in the function.

Make sure to remove the unused `ValidateColumnFunction` type above :) 

```suggestion
function validateAllowedColumns(validColumns: string[]): ValidateColumnValueFunction {
  return function ({name}) {
    return validColumns.includes(name);
  };
}
```